common/src/main/spark-3.3/org/apache/comet/shims/ShimBatchReader.scala

```diff
@@ -24,31 +24,13 @@ import org.apache.spark.sql.execution.datasources.PartitionedFile
 
 object ShimBatchReader {
 
-  // TODO: remove after dropping Spark 3.3 support and directly call PartitionedFile
   def newPartitionedFile(partitionValues: InternalRow, file: String): PartitionedFile =
-    classOf[PartitionedFile].getDeclaredConstructors
-      .map(c =>
-        c.getParameterCount match {
-          case 5 =>
-            c.newInstance(
-              partitionValues,
-              file,
-              Long.box(-1), // -1 means we read the entire file
-              Long.box(-1),
-              Array.empty[String])
-          case 7 =>
-            c.newInstance(
-              partitionValues,
-              c.getParameterTypes()(1)
-                .getConstructor(classOf[String])
-                .newInstance(file)
-                .asInstanceOf[AnyRef],
-              Long.box(-1), // -1 means we read the entire file
-              Long.box(-1),
-              Array.empty[String],
-              Long.box(0),
-              Long.box(0))
-        })
-      .head
-      .asInstanceOf[PartitionedFile]
+    PartitionedFile(
+      partitionValues,
+      file,
+      Long.box(-1), // -1 means we read the entire file
+      Long.box(-1),
+      Array.empty[String],
+      Long.box(0),
+      Long.box(0))
```

Comment:
hmmm i thought Spark 3.3 uses 5 arguments, but I could be wrong.

Reply:
3.2 uses 5 arguments. 3.3 and 3.4 both use 7 arguments.